src/App.tsx

```diff
@@ -15,47 +16,140 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType) {
+    visibleGoods.sort((prevGood, currentGood) => {
+      switch (sortType) {
+        case SortType.ALPABET:
+          return prevGood.localeCompare(currentGood);
+
+        case SortType.LENGTH:
+          return prevGood.length - currentGood.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed]
+    = useState<State['isReversed']>(false);
```

Comment:
`<State['isReversed']>` redundant here.
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```